src/tracks/playabletrack/wavetrack/ui/WaveformView.cpp

```diff
@@ -846,14 +840,14 @@ void DrawClipWaveform(TrackPanelDrawingContext &context, size_t channel,
          if (!showIndividualSamples) {
             std::vector<double> vEnv2(rectPortion.width);
             double *const env2 = &vEnv2[0];
-            CommonChannelView::GetEnvelopeValues(envelope,
-                tOffset,
+            CommonChannelView::GetEnvelopeValues(
+               *clip.GetEnvelope(), playStartTime,
 
-                 // PRL: change back to make envelope evaluate only at sample
-                 // times and then interpolate the display
-                 0, // 1.0 / rate,
+               // PRL: change back to make envelope evaluate only at sample
+               // times and then interpolate the display
```

Comment:
Below, change `rate` to `sampleRate` in commented-out code.  Sample rate NOT corrected for stretch ratio would be right because we assume the envelope already stretches its time coordinates.